File: JavaToKotlin/apps(kotlin)/src/main/kotlin/io/dustin/apps/board/api/usecase/qna/answer/DeleteAnswerUseCase.java
package io.dustin.apps.board.api.usecase.qna.answer;

import io.dustin.apps.board.domain.qna.answer.service.ReadAnswerService;
import io.dustin.apps.board.domain.qna.answer.service.WriteAnswerService;
import io.dustin.apps.board.domain.qna.answer.model.Answer;
import io.dustin.apps.board.domain.qna.answer.model.dto.AnswerDto;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;
import org.springframework.web.server.ResponseStatusException;

import java.security.Principal;

@Service
@RequiredArgsConstructor
public class DeleteAnswerUseCase {

    private final ReadAnswerService readAnswerService;
    private final WriteAnswerService writeAnswerService;

    @Transactional
    public AnswerDto execute(Long adminId, Long answerId) {
        Answer answer = readAnswerService.findById(answerId);
        if(!answer.getAdminId().equals(adminId)) {
            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "삭제 권한이 없습니다.");
        }
        writeAnswerService.delete(answer);
        return AnswerDto.from(answer);
    }

}

File: apps/src/main/kotlin/io/dustin/apps/board/api/usecase/qna/answer/DeleteAnswerUseCase.kt
package io.dustin.apps.board.api.usecase.qna.answer

import io.dustin.apps.board.api.qna.request.command.AnswerDeleteCommand
import io.dustin.apps.board.domain.qna.answer.model.dto.AnswerDto
import io.dustin.apps.board.domain.qna.answer.service.ReadAnswerService
import io.dustin.apps.board.domain.qna.answer.service.WriteAnswerService
import io.dustin.apps.common.exception.DataNotFoundException
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
import javax.naming.AuthenticationException

@Service
class DeleteAnswerUseCase(
    private val readAnswerService: ReadAnswerService,
    private val writeAnswerService: WriteAnswerService
) {
    @Transactional
    fun execute(command: AnswerDeleteCommand) {
        val answer = readAnswerService.findByIdOrThrow(command.answerId)
        if(answer.adminId == command.adminId){
            writeAnswerService.delete(answer)
        }
        else {
            return throw AuthenticationException("권한이 없습니다.")
        }
    }
}
